app/src/main/java/org/xbmc/kore/ui/MoviesActivity.java

```diff
@@ -168,31 +186,42 @@ private void setupActionBar(String movieTitle) {
     /**
      * Callback from movielist fragment when a movie is selected.
      * Switch fragment in portrait
-     * @param movieId Movie selected
-     * @param movieTitle Title
+     * @param vh ViewHolder holding movie info of item clicked
      */
     @TargetApi(21)
-    public void onMovieSelected(int movieId, String movieTitle) {
-        selectedMovieId = movieId;
-        selectedMovieTitle = movieTitle;
-
-        MovieDetailsFragment movieDetailsFragment = MovieDetailsFragment.newInstance(movieId);
-        FragmentTransaction fragTrans = getSupportFragmentManager().beginTransaction();
+    public void onMovieSelected(MovieListFragment.ViewHolder vh) {
+        selectedMovieTitle = vh.movieTitle;
+        selectedMovieId = vh.movieId;
 
         // Set up transitions
         if (Utils.isLollipopOrLater()) {
+            MovieDetailsFragment movieDetailsFragment = MovieDetailsFragment.newInstance(vh);
+            FragmentTransaction fragTrans = getSupportFragmentManager().beginTransaction();
+
             movieDetailsFragment.setEnterTransition(TransitionInflater
                     .from(this)
                     .inflateTransition(R.transition.media_details));
-            movieDetailsFragment.setReturnTransition(null);
```

Comment:
Not to be removed. 
Removing this causes the details fragment to slide down on the return transition, which is very distracting and looks odd, with the list fragment fading into view.
I think that having null for the return transition gives the best effect, but a fade out also works (though it somewhat interferes visually with the fade in of the list fragment).